File: frontend/src/api/modelSubscription.ts
import request from './request'

export interface ModelSubscription {
  id: number
  modelId: number
  userId: number
  tenantId?: string
  appId?: string
  status: number
  createTime?: string
  updateTime?: string
}

export const subscribeModel = (modelId: number, userId: number, tenantId?: string, appId?: string) => {
  return request.post<any, boolean>('/api/admin/model-subscriptions/subscribe', null, {
    params: { modelId, userId, tenantId, appId },
  })
}

export const unsubscribeModel = (modelId: number, userId: number, tenantId?: string, appId?: string) => {
  return request.post<any, boolean>('/api/admin/model-subscriptions/unsubscribe', null, {
    params: { modelId, userId, tenantId, appId },
  })
}

export const getMySubscriptions = (userId: number, tenantId?: string, appId?: string) => {
  return request.get<any, ModelSubscription[]>('/api/admin/model-subscriptions/my', {
    params: { userId, tenantId, appId },
  })
}

File: gateway-admin/src/main/java/com/aigateway/admin/entity/ModelSubscription.java
package com.aigateway.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 模型订阅关系
 */
@Data
@TableName("model_subscription")
public class ModelSubscription {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long modelId;

    private Long userId;

    private String tenantId;

    private String appId;

    /**
     * 订阅状态：1=订阅，0=取消
     */
    private Integer status;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

